pkg/oc/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,158 @@
+	kapi "k8s.io/kubernetes/pkg/apis/core"
+	rbacv1helper "k8s.io/kubernetes/pkg/apis/rbac/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/oc/cli/util/clientcmd"
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	securitytypedclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	internalMigrateSCCShort = "Converts SCCs to similar PSPs"
+	internalMigrateSCCLong  = templates.LongDesc(`
+		Converts SecurityContextConstraints to equivalent PodSecurityPolicy objects.
+
+		NOTE: it does not modify anything by default.`)
+
+	internalMigrateSCCExample = templates.Examples(`
+		# Output to the console PodSecurityPolicy objects
+		%[1]s`)
+)
+
+type migrateSCCOptions struct {
+	sccClient securitytypedclient.SecurityContextConstraintsInterface
+}
+
+func NewCmdMigrateSCC(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &migrateSCCOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   internalMigrateSCCShort,
+		Long:    internalMigrateSCCLong,
+		Example: fmt.Sprintf(internalMigrateSCCExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run(cmd, f, out))
+		},
+	}
+	kcmdutil.AddPrinterFlags(cmd)
+	return cmd
+}
+
+func (o *migrateSCCOptions) Complete(name string, f *clientcmd.Factory, c *cobra.Command, args []string) error {
+	securityClient, err := f.OpenshiftInternalSecurityClient()
+	if err != nil {
+		return err
+	}
+	o.sccClient = securityClient.Security().SecurityContextConstraints()
+	return nil
+}
+
+func (o migrateSCCOptions) Validate() error {
+	if o.sccClient == nil {
+		return errors.New("a SCC client is required")
+	}
+	return nil
+}
+
+func (o migrateSCCOptions) Run(cmd *cobra.Command, f *clientcmd.Factory, out io.Writer) error {
+	sccs, err := o.sccClient.List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	list := &kapi.List{}
+	for _, scc := range sccs.Items {
+		list.Items = append(list.Items, convertSccToPsp(&scc))
+		list.Items = append(list.Items, convertSccToClusterRole(&scc))
+		binding := convertSccToClusterRoleBinding(&scc)
+		if binding != nil {
+			list.Items = append(list.Items, binding)
+		}
+	}
+
+	mapper, _ := f.Object()
+	fn := cmdutil.VersionedPrintObject(f.PrintObject, cmd, mapper, out)
+	if err := fn(list); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func convertSccToPsp(scc *securityapi.SecurityContextConstraints) *extensions.PodSecurityPolicy {
+	// TODO: convert SCC.Priority *int32 (ERR: PSP doesn't have such field)
+	// TODO: convert SCC.AllowHostPorts bool (TODO: how to convert bool to []HostPortRange{Min:, Max:}?)
```

Comment:
false == empty host port range, true == host port range from PORT_MIN-PORT_MAX

Reply:
... where PORT_MIN is 0 and PORT_MAX is 65535, correct?